src/main/kotlin/rys/ajaxpetproject/repository/impl/mongoTemplate/ChatMongoTemplate.kt

```diff
@@ -0,0 +1,31 @@
+package rys.ajaxpetproject.repository.impl.mongoTemplate
+
+import org.bson.types.ObjectId
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.query.Criteria
+import org.springframework.data.mongodb.core.query.Query
+import org.springframework.stereotype.Repository
+import rys.ajaxpetproject.model.MongoChat
+import rys.ajaxpetproject.repository.ChatDAO
+
+@Repository
+class ChatMongoTemplate(private val mongoTemplate: MongoTemplate) : ChatDAO {
```

Comment:
```suggestion
class ChatMongoRepository(private val mongoTemplate: MongoTemplate) : ChatDao {
```

Why did you decide to name your repositories as templates? just curious

Reply:
Because they work with MongoTemplate. Thought this way className will be more descriptive. 
@Repository isn't enough to tell that this class - repository?
Done

Reply:
I can extrapolate this logic like "if a service works with ChatMongoTemplate - then it is ChatServiceMongoTemplate. and if a controller works with ChatServiceMongoTemplate - then it is ChatControllerMongoTemplate" :)
My point is to name classes according to WHAT they do, not HOW they do it. If you refactor the repository to use spring data (or some other way) instead of mongo template, will it change class purpose?

Reply:
Good point. Got it